File: searchingAlgo/DepthFirstSearch/depthFirstSearch.cs
using System; 
using System.Collections.Generic; 
class Graph 
{ 
    private int V;
    private List<int>[] adj; 
    Graph(int v) 
    { 
        V = v; 
        adj = new List<int>[v]; 
        for (int i = 0; i < v; ++i) 
            adj[i] = new List<int>(); 
    } 
    void AddEdge(int v, int w) 
    { 
        adj[v].Add(w);
    } 
    void DFSUtil(int v, bool[] visited) 
    { 
        visited[v] = true; 
        Console.Write(v + " "); 
        List<int> vList = adj[v]; 
        foreach (var n in vList) 
        { 
            if (!visited[n]) 
                DFSUtil(n, visited); 
        } 
    } 

    void DFS(int v) 
    {   
        bool[] visited = new bool[V]; 
        DFSUtil(v, visited); 
    } 
    public static void Main(String[] args) 
    { 
        Graph g = new Graph(4); 
        g.AddEdge(0, 1); 
        g.AddEdge(0, 2); 
        g.AddEdge(1, 2); 
        g.AddEdge(2, 0); 
        g.AddEdge(2, 3); 
        g.AddEdge(3, 3); 
        Console.WriteLine("Following is Depth First Traversal " + 
                          "(Starting from vertex 2)"); 
        g.DFS(2); 
        Console.ReadKey(); 
    } 
} 

File: searchingAlgo/DepthFirstSearch/depthFirstSearch.js
// Graph class for the Algorithm
class Graph {
  constructor(v) {
    this.V = v;
    this.adj = Array(v);
    for (let i = 0; i < v; ++i) {
      this.adj[i] = [];
    }
  }

  addEdge(v, w) {
    this.adj[v].push(w);
  }

  DFSUtil(v, visited) {
    visited[v] = true;
    console.log(v + " ");
    let vList = this.adj[v];
    for (let n in vList) {
      if (!visited[n]) this.DFSUtil(n, visited);
    }
  }

  DFS(v) {
    let visited = new Array(this.V);
    this.DFSUtil(v, visited);
  }
}

function main() {
  let g = new Graph(4);
  g.addEdge(0, 1);
  g.addEdge(0, 2);
  g.addEdge(1, 2);
  g.addEdge(2, 0);
  g.addEdge(2, 3);
  g.addEdge(3, 3);
  console.log(
    "The Following is Depth First Traversal " + "(Starting from vertex 2)"
  );
  g.DFS(2);
}

main();
